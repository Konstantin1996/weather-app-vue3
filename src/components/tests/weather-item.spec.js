
import { shallowMount } from '@vue/test-utils';
import WeatherItem from '../weather-item.vue';

describe('WeatherItem', () => {
    const renderWrapper = (props) => {
        return shallowMount(WeatherItem, { propsData: { ...props } });
    };

    it('should display name, region, country, temperature, condition', () => {
        const wrapper = renderWrapper({
            name: 'Custom name',
            region: 'custom region',
            country: 'country',
            tempInCelsius: 45.3,
            weatherCondition: 'Sunny',
            weatherIcon: 'https://t4.ftcdn.net/jpg/03/63/38/53/360_F_363385318_FGumiZOnWi4vRuaFJ2Df0KpidbzLp2fu.jpg'
        });

        expect(wrapper.text()).toBe('Name: Custom nameRegion: custom regionCountry: countryTemperature: 45.3 Â°CCondition: Sunny');
        expect(wrapper.element).toMatchSnapshot();
    })
})