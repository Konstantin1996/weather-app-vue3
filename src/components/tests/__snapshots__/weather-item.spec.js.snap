// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherItem should display name, region, country, temperature, condition 1`] = `
<div
  data-v-app=""
>
  
  <p>
    Name: Custom name
  </p>
  <p>
    Region: custom region
  </p>
  <p>
    Country: country
  </p>
  <p>
    Temperature: 45.3 Â°C
  </p>
  <p>
    Condition: Sunny
  </p>
  <img
    alt="weather-img"
    src="https://t4.ftcdn.net/jpg/03/63/38/53/360_F_363385318_FGumiZOnWi4vRuaFJ2Df0KpidbzLp2fu.jpg"
  />
  
</div>
`;
